pkg/private/secant/sign.go

```diff
@@ -175,3 +185,29 @@ func (r skipSameSignatures) Find(signatures oci.Signatures, o oci.Signature) (oc
 
 	return nil, nil
 }
+
+type signOptions struct {
+	rekorLimiter *rate.Limiter
+}
+
+type SignOption (func(*signOptions) error)
+
+func makeSignOptions(opts []SignOption) (*signOptions, error) {
+	o := &signOptions{
+		rekorLimiter: defaultRekorLimiter,
+	}
+	for _, opt := range opts {
+		if err := opt(o); err != nil {
+			return nil, err
+		}
+	}
+	return o, nil
+}
+
+// WithSignRekorLimiter rate limits calls to Rekor
+func WithSignRekorLimiter(limiter *rate.Limiter) AttestOption {
```

Comment:
I would avoid functional options if we can.

Reply:
Are you suggesting to avoid making this configurable (using a hard-coded default rate limiter kind of like it is now), or just avoiding functional arguments?

This method already has a lengthy set of arguments, so I was trying to avoid adding another mandatory one to the callsite.